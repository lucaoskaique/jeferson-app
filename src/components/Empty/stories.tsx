import { Story, Meta } from '@storybook/react'

import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty,
  parameters: {
    layout: 'fullscreen',
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />
Default.args = {
  title: 'Empty',
  description: 'No data',
  hasLink: true
}
