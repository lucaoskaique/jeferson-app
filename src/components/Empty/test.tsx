import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

const props = {
  title: 'No results',
  description: 'A simple description'
}

describe('<Empty />', () => {
  it('should render Empty', () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('heading', { name: /No results/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/a simple description/i)).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /go back to home/i })
    ).toHaveAttribute('href', '/')

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render Empty without link', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole('link', { name: /go back to home/i })
    ).not.toBeInTheDocument()
  })
})
