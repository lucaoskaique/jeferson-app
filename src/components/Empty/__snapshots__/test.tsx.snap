// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render Empty 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: linear-gradient(90deg,#ffbf3a -40%,#d90726 185.31%);
}

.c3:focus {
  outline: 1px dashed;
}

.c3:hover {
  background: linear-gradient(90deg,#72c042 -99.98%,#0092d3 193.73%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  color: #EC6330;
  font-size: 2.8rem;
}

.c2 {
  color: #030517;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<body>
  <div>
    <main
      class="c0"
    >
      <h2
        class="c1"
      >
        No results
      </h2>
      <p
        class="c2"
      >
        A simple description
      </p>
      <a
        class="c3"
        href="/"
      >
        <span>
          Go back to Home
        </span>
      </a>
    </main>
  </div>
</body>
`
